### PR TITLE
Let config accept both int and float as float type

### PR DESCRIPTION
**Pull request by cydrain**

### CODE REVIEW

File: include/knowhere/config.h [hunk 1]
```diff
@@ -299,7 +299,7 @@ class Config {
                     *ptr->val = ptr->default_val.value();
                     continue;
                 }
-                if (!json[it.first].is_number_float()) {
+                if (!json[it.first].is_number()) {
```
Comment (PwzXxm) [inline include/knowhere/config.h:302]: The arithmetic overflow check below becomes invalid, and all float values in the config can be float/int, is this acceptable? @liliu-z

Comment (liliu-z) [reply include/knowhere/config.h:302]: Looks good to me. cc @Presburger as the original author.

Comment (Presburger) [reply include/knowhere/config.h:302]: Looks good.

**Review by Presburger — APPROVED** [review]

**Comment by Presburger** [comment]
\lgtm

**Pull request merged**